src/utils/AuthUtils.js

```diff
@@ -172,6 +175,37 @@ const createUser = async user => {
   await auth.signOut();
 };
 
+// Updates user information in PNP DB
+const updateUserInDB = async (user, id) => {
+  const { firstName, lastName, email, phoneNumber, role } = user;
+  try {
+    await PNPBackend.put(`/users/${id}`, {
+      firstName,
+      lastName,
+      email,
+      phoneNumber,
+      role,
+      id,
+    });
+  } catch (err) {
+    throw new Error(err.message);
+  }
+};
+
+/**
+ * Updates user password in Firebase after verifying that current password is correct
+ * so that the user stays logged in after a password change
+ */
+const updateFirebaseUser = async user => {
+  const { email, currentPassword, newPassword } = user;
+  if (currentPassword && newPassword) {
+    const credentials = EmailAuthProvider.credential(email, currentPassword);
+    await reauthenticateWithCredential(auth.currentUser, credentials);
+    await updatePassword(auth.currentUser, newPassword);
+  }
+  await updateUserInDB(user, auth.currentUser.uid);
+};
+
```

Comment:
As mentioned in the comment in `UserDetails.jsx`, we should change to function. It might be easiest to reuse the `PUT` backend route and do a conditional check to see if a password was passed in the body to call firebase to update it.